packages/http-client-python/emitter/src/emitter.ts

```diff
@@ -85,51 +87,113 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
     });
     return;
   }
-
-  addDefaultOptions(sdkContext);
   const yamlPath = await saveCodeModelAsYaml("python-yaml-path", yamlMap);
-  let venvPath = path.join(root, "venv");
-  if (fs.existsSync(path.join(venvPath, "bin"))) {
-    venvPath = path.join(venvPath, "bin", "python");
-  } else if (fs.existsSync(path.join(venvPath, "Scripts"))) {
-    venvPath = path.join(venvPath, "Scripts", "python.exe");
-  } else {
-    throw new Error("Virtual environment doesn't exist.");
-  }
-  const commandArgs = [
-    venvPath,
-    `${root}/eng/scripts/setup/run_tsp.py`,
-    `--output-folder=${outputDir}`,
-    `--cadl-file=${yamlPath}`,
-  ];
+  addDefaultOptions(sdkContext);
   const resolvedOptions = sdkContext.emitContext.options;
+  const commandArgs: Record<string, string> = {};
   if (resolvedOptions["packaging-files-config"]) {
     const keyValuePairs = Object.entries(resolvedOptions["packaging-files-config"]).map(
       ([key, value]) => {
         return `${key}:${value}`;
       },
     );
-    commandArgs.push(`--packaging-files-config='${keyValuePairs.join("|")}'`);
+    commandArgs["packaging-files-config"] = keyValuePairs.join("|");
     resolvedOptions["packaging-files-config"] = undefined;
   }
   if (
     resolvedOptions["package-pprint-name"] !== undefined &&
     !resolvedOptions["package-pprint-name"].startsWith('"')
   ) {
-    resolvedOptions["package-pprint-name"] = `"${resolvedOptions["package-pprint-name"]}"`;
+    resolvedOptions["package-pprint-name"] = `${resolvedOptions["package-pprint-name"]}`;
   }
 
   for (const [key, value] of Object.entries(resolvedOptions)) {
-    commandArgs.push(`--${key}=${value}`);
+    commandArgs[key] = value;
   }
   if (sdkContext.arm === true) {
-    commandArgs.push("--azure-arm=true");
+    commandArgs["azure-arm"] = "true";
   }
   if (resolvedOptions.flavor === "azure") {
-    commandArgs.push("--emit-cross-language-definition-file=true");
+    commandArgs["emit-cross-language-definition-file"] = "true";
   }
-  commandArgs.push("--from-typespec=true");
+  commandArgs["from-typespec"] = "true";
+
   if (!program.compilerOptions.noEmit && !program.hasError()) {
-    execSync(commandArgs.join(" "));
+    // if not using pyodide and there's no venv, we try to create venv
+    if (!resolvedOptions["use-pyodide"] && !fs.existsSync(path.join(root, "venv"))) {
+      try {
+        await runPython3("./eng/scripts/setup/install.py");
+        await runPython3("./eng/scripts/setup/prepare.py");
+      } catch (error) {
+        if (
+          error instanceof Error &&
+          error.message.includes("Failed to find compatible python version.")
```

Comment:
The error message is same in system-requirements.ts so maybe it is better to define a global constant so that we don't need to change both if needed.